### PR TITLE
Support urls with dns zone

### PR DESCRIPTION
**Pull request by ohadbitt**
Also renamed trusted_endpoints package to trustedendpoints as this change sat in my env for a while after not pushing it

### CODE REVIEW

File: kusto/test/etoe/etoe_test.go [hunk 1]
```diff
@@ -20,6 +15,12 @@ import (
 	"time"
 	"unicode"
 
+	"github.com/Azure/azure-kusto-go/kusto/utils"
```
Comment (AsafMah) [inline kusto/test/etoe/etoe_test.go:18]: why?

File: kusto/ingest/internal/resources/resources.go [hunk 2]
```diff
@@ -63,12 +64,23 @@ func parse(uri string) (*URI, error) {
...
 	}
 
-	v := &URI{
-		u:          u,
-		account:    hostSplit[0],
-		objectType: hostSplit[1],
-		objectName: strings.TrimLeft(u.EscapedPath(), "/"),
-		sas:        u.Query(),
+	var v *URI
+	if len(hostSplit) == 5 {
```
Comment (AsafMah) [inline kusto/ingest/internal/resources/resources.go:68]: that's the only way to check? the zone can't happen to have 5 segments? is it what we do in other sdks?

**Review by AsafMah — CHANGES_REQUESTED** [review]

File: kusto/test/etoe/etoe_test.go (hunk 1 above)
Comment (ohadbitt) [reply kusto/test/etoe/etoe_test.go:18]: revert

File: kusto/ingest/internal/resources/resources.go (hunk 2 above)
Comment (ohadbitt) [reply kusto/ingest/internal/resources/resources.go:68]: This is what we do for example- you see we use the objectName

	
	return azqueue.NewServiceURL(*service, p).NewQueueURL(queue.ObjectName()).NewMessagesURL(), nil

Comment (AsafMah) [reply kusto/ingest/internal/resources/resources.go:68]: https://github.com/Azure/azure-storage-blob-go is deprecated - see the readme (if that's what you meant)

I think they might have added methods to make it a bit easier recently.

Either way I'll resolve for now

**Pull request merged**